app/llama_sensei/backend/qa/generate_answer.py

```diff
@@ -1,31 +1,48 @@
+import time
 from datetime import datetime
 
+import numpy as np
+import torch
 from datasets import Dataset
+from langchain_community.utilities import DuckDuckGoSearchAPIWrapper
 from langchain_groq import ChatGroq
 from llama_sensei.backend.add_courses.vectordb.document_processor import (
     DocumentProcessor,
 )
 from ragas import evaluate
 from ragas.metrics import faithfulness
+from sentence_transformers import SentenceTransformer
+from sklearn.metrics.pairwise import cosine_similarity
 
 MODEL = "llama3-70b-8192"
+# device = "cuda" if torch.cuda.is_available() else "cpu"
+# EMBEDDING_LLM = SentenceTransformer("all-MiniLM-L12-v2", trust_remote_code=True).to(device)
+EMBEDDING_LLM = "all-MiniLM-L12-v2"
 
 
 class GenerateRAGAnswer:
     def __init__(self, query: str, course: str, model=MODEL):
         self.query = query
         self.course = course
+        device = "cuda" if torch.cuda.is_available() else "cpu"
```

Comment:
`device` has no duty in initialization here, move it out just like `EMBEDDING_LLM`